src/runtime/task/mod.rs

```diff
@@ -265,3 +287,146 @@ impl Debug for TaskSet {
         write!(f, "}}")
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct VectorClock {
+    pub(crate) time: SmallVec<[u32; DEFAULT_INLINE_TASKS]>,
```

Comment:
Is it worth introducing a wrapper type for timestamps, `Timestamp(u32)` or something?

Reply:
Feels a bit overkill, since most of the time we're comparing and updating entire clocks.  We only look at individual timestamps in the VectorClock impl and in test code.